chain/events/filter/index.go

```diff
@@ -407,6 +429,63 @@ func (ei *EventIndex) migrateToVersion5(ctx context.Context) error {
 	return nil
 }
 
+func (ei *EventIndex) migrateToVersion6(ctx context.Context) error {
+	now := time.Now()
+
+	tx, err := ei.db.BeginTx(ctx, nil)
+	if err != nil {
+		return xerrors.Errorf("begin transaction: %w", err)
+	}
+	defer func() { _ = tx.Rollback() }()
+
+	stmtCreateTableEventsSeen, err := tx.PrepareContext(ctx, createTableEventsSeen)
+	if err != nil {
+		return xerrors.Errorf("prepare stmtCreateTableEventsSeen: %w", err)
+	}
+	_, err = stmtCreateTableEventsSeen.ExecContext(ctx)
+	if err != nil {
+		return xerrors.Errorf("create table events_seen: %w", err)
+	}
+
+	_, err = tx.ExecContext(ctx, createIndexEventsSeenHeight)
+	if err != nil {
+		return xerrors.Errorf("create index events_seen_height: %w", err)
+	}
+	_, err = tx.ExecContext(ctx, createIndexEventsSeenTipsetKey)
+	if err != nil {
+		return xerrors.Errorf("create index events_seen_tipset_key: %w", err)
+	}
+
+	// INSERT an entry in the events_seen table for all epochs we do have events for in our DB
+	_, err = tx.ExecContext(ctx, `
+    INSERT INTO events_seen (height, tipset_key, tipset_key_cid)
+    SELECT e.height, e.tipset_key, e.tipset_key_cid
+    FROM event e
+    WHERE NOT EXISTS (
+        SELECT 1 FROM events_seen es
+        WHERE es.height = e.height AND es.tipset_key = e.tipset_key AND es.tipset_key_cid = e.tipset_key_cid
```

Comment:
an alternative would be to find the min and max and fill in all between those two as being seen, that might yield a a state that's much more likely to be valid; there's no good reason to have blanks between min and max, but there are good reasons to have missing events, (this can been seen particularly on calibnet)

Reply:
@rvagg min and max of what ? The problem is that for epochs that do not have events, we simply do not have any entry for them in any of our event tabls.

Reply:
I was thinking min and max height; start from `MIN(height)` and add an entry for every one until `MAX(height)` .. but I suppose you need tipset_key_cid for each of those too so you'd have to go looking for it - but this could become a chain walk operation too -- find min and max, walk chain between and add an entry for each one.

Maybe this isn't worth thinking too hard about since it's a migration but I _think_ doing a walk and including _every_ epoch in that range is much more likely to be accurate than just filling in the ones we have events for. Because we should have an entry in the table even for tipsets that have no events because that's part of what we're solving here - no events in the tipset, vs no events in the database but maybe tipset has events?